### PR TITLE
379794 - Log user requests with the user name

### PR DESCRIPTION
**Pull request by alexluckett**

### CODE REVIEW

File: src/plugins/log-requests.js [hunk 1]
```diff
@@ -9,7 +9,25 @@ export const logRequests = {
...
+        if (
+          user &&
+          'preferred_username' in user &&
+          typeof user.preferred_username === 'string'
+        ) {
+          userPrefix = ` [${user.preferred_username}] `
+        }
+      }
+
+      return `[response]${userPrefix} ${request.method} ${request.path} ${request.raw.res.statusCode} (${responseTime}ms)`
```
Comment (alexluckett) [inline src/plugins/log-requests.js:29]: this is the default hapi logger format with `${userPrefix}` added

File: src/plugins/auth.js [hunk 2]
```diff
@@ -36,20 +36,29 @@ export const auth = {
...
+          if (!user) {
+            return {
+              isValid: false
+            }
+          }
+
+          const groups = Array.isArray(user.groups) ? user.groups : []
+
+          logger.debug(
+            `User ${user.preferred_username}: validating against groups: ${groups.join(', ')}`
```
Comment (colinrotherham) [inline src/plugins/auth.js:49]: Are we alright with how this line logs empty `[]` groups?

**Review by davidjamesstone — APPROVED** [review]

File: src/plugins/auth.js [hunk 3]
```diff
@@ -36,20 +36,29 @@ export const auth = {
...
-          const groups = Array.isArray(user?.groups) ? user.groups : []
 
-          logger.debug(`Validating user against groups: ${groups.join(', ')}`)
+          if (!user) {
+            return {
+              isValid: false
+            }
+          }
+
+          const groups = Array.isArray(user.groups) ? user.groups : []
```
Comment (colinrotherham) [inline src/plugins/auth.js:46]: Is groups sometimes not an array? Default is good

```suggestion
          const { groups = [] } = user
```

File: src/plugins/auth.js [hunk 4]
```diff
@@ -36,20 +36,29 @@ export const auth = {
...
+
+          logger.debug(
+            `User ${user.preferred_username}: validating against groups: ${groups.join(', ')}`
+          )
 
           if (!groups.includes(roleEditorGroupId)) {
             logger.warn(
-              `User failed authorisation. "${roleEditorGroupId}" not in groups.`
+              `User ${user.preferred_username}: failed authorisation. "${roleEditorGroupId}" not in groups.`
```
Comment (colinrotherham) [inline src/plugins/auth.js:54]: Are we alright with how this line logs empty `preferred_username`?

File: src/plugins/auth.js (hunk 2 above)
Comment (alexluckett) [reply src/plugins/auth.js:49]: Enhancement in PR https://github.com/DEFRA/forms-manager/pull/176.

Commit https://github.com/DEFRA/forms-manager/pull/176/commits/71fd2590aff2f04b02aebd083d3e2ae23c975a93.

File: src/plugins/auth.js (hunk 3 above)
Comment (alexluckett) [reply src/plugins/auth.js:46]: Enhancement in PR https://github.com/DEFRA/forms-manager/pull/176.

Commit https://github.com/DEFRA/forms-manager/pull/176/commits/45a41ff4cf9f7b2c8bfcf5de352d031b1cf54c28.

File: src/plugins/auth.js (hunk 4 above)
Comment (alexluckett) [reply src/plugins/auth.js:54]: Enhancement in PR https://github.com/DEFRA/forms-manager/pull/176.

Fixed in commit https://github.com/DEFRA/forms-manager/pull/176/commits/45a41ff4cf9f7b2c8bfcf5de352d031b1cf54c28.

File: src/plugins/log-requests.js [hunk 5]
```diff
@@ -9,7 +9,25 @@ export const logRequests = {
   plugin: hapiPino,
   options: {
     ...loggerOptions,
-    log4xxResponseErrors: true
+    log4xxResponseErrors: true,
+    logRequestComplete: true,
+    customRequestCompleteMessage(request, responseTime) {
+      let userPrefix = ''
```
[the remark this answers is not in the document]
Comment (alexluckett) [reply src/plugins/log-requests.js:15]: Enhancement in PR https://github.com/DEFRA/forms-manager/pull/176.

Fixed in commit https://github.com/DEFRA/forms-manager/commit/1126f8e7ebcf9203e0650622bb6c74b00501a3db.

File: src/plugins/log-requests.js [hunk 6]
```diff
@@ -9,7 +9,25 @@ export const logRequests = {
...
+
+      if (request.auth.isAuthenticated) {
+        const { user } = request.auth.credentials
```
[the remark this answers is not in the document]
Comment (alexluckett) [reply src/plugins/log-requests.js:18]: Enhancement in PR https://github.com/DEFRA/forms-manager/pull/176.

Fixed in commit https://github.com/DEFRA/forms-manager/pull/176/commits/1126f8e7ebcf9203e0650622bb6c74b00501a3db.

**Pull request merged**